src/classifier/functions.py

```
@@ -0,0 +1,667 @@
+    y_val : pd.Series con etiquetas de validación.
+    params : dict, hiperparámetros para XGBClassifier (opcional).
+
+    Returns:
+    modelo : XGBClassifier entrenado.
+    """
+    from xgboost import XGBClassifier
+    from sklearn.preprocessing import LabelEncoder
+
+    le = LabelEncoder()
+    y_train_enc = le.fit_transform(y_train)
+    y_val_enc = le.transform(y_val)
```

Ruff (0.15.1)

[error] Local variable `y_val_enc` is assigned to but never used

Remove assignment to unused variable `y_val_enc`

(F841)